File: src/model/statement/IfStmt.java
package model.statement;

import model.Expression;
import model.ProgState;
import utils.IHeap;
import utils.ISymbolTable;

public class IfStmt implements Statement{
	private Expression exp;
	private Statement stm1,stm2;
	
	public IfStmt(Expression ex,Statement st1,Statement st2){
		exp=ex;
		stm1=st1;
		stm2=st2;
	}
	
	@Override
	public ProgState execute(ProgState p){
		ISymbolTable <String,Integer> t = p.getSymbolTable();
		IHeap<Integer,Integer> hp = p.getHeap();
		int result = exp.evaluate(t,hp);
		if(result==1)
			p.getExeStack().push(stm1);
		else p.getExeStack().push(stm2);
		return null;
	}
		
	public String toString(){
		return "if ("+exp+") then "+stm1+"; else "+stm2;
	}
}

File: ToyLanguageInterpretorC#/ToyLanguageC/model/IfStmt.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ToyLanguageC.utils;

namespace ToyLanguageC.model
{
    class IfStmt : Statement
    {
        Expression exp;
        Statement stm1;
        Statement stm2;

        public IfStmt(Expression exp1,Statement stm1,Statement stm2)
        {
            this.exp = exp1;
            this.stm1 = stm1;
            this.stm2 = stm2;
        }

        public ProgState execute(ProgState state)
        {
            ExeStack<Statement> stack = state.getExeStack;
            SymbolTable<string, int> table = state.getSymbolTable;
            int val = exp.evaluate(table);
            if (val != 0)
                stack.push(stm1);
            else stack.push(stm2);
            return state;
        }

        public string toString()
        {
            return "IF(" + exp.toString() + ") THEN(" + stm1.toString()+ ")ELSE(" + stm2.toString() + ")"; ;
        }
    }
}
